### PR TITLE
bucketize

### LINKED ISSUE #14083: bucketize

### PR DESCRIPTION
**Pull request by Aneesh02**
Closes #14083

### CODE REVIEW

**Comment by ZiadAmerr** [comment]
Hi @Aneesh02, thanks for your contribution. I will leave a few comments on the code and we can continue with the merge after that :)

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 1]
```diff
@@ -1292,3 +1292,40 @@ def test_torch_diag(
         input=values[0],
         diagonal=diagonal,
     )
+    
+ 
+@handle_frontend_test(
+    fn_tree="torch.bucketize",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1300]: Why are we restricting the arrays generated to float only? I suggest using `valid` instead

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 2]
```diff
@@ -1292,3 +1292,40 @@ def test_torch_diag(
...
     )
+    
+ 
+@handle_frontend_test(
+    fn_tree="torch.bucketize",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        shared_dtype=True,
+        min_num_dims=1,
+        max_num_dims=1,
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1303]: Why are we restricting the number of dims to 1 only?

The following is an example from PyTorch's documentation

```
>>> boundaries = torch.tensor([1, 3, 5, 7, 9])
>>> v = torch.tensor([[3, 6, 9], [3, 6, 9]])
>>> torch.bucketize(v, boundaries)
```

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 3]
```diff
@@ -1292,3 +1292,40 @@ def test_torch_diag(
...
+    ),
+    boundaries=st.lists(st.integers(1, 10), min_size=1, max_size=10)),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1305]: `boundaries` was specified by torch's documentation as:

> 1-D tensor, must contain a strictly increasing sequence, or the return value is undefined.

But `st.lists(...)` doesn't necessarily generate an increasing sequence, it is a good approach to exhaustively test your function, but the probability that the strategy does generate a valid increasing sequence is very low as it randomly samples numbers from a list of integers. So all our test cases might be falling under the _"return value is undefined"_ criterion.

**Comment by ZiadAmerr** [comment]
You should also make sure that the linting errors are passing, because I see some linting errors in your code.

Simply run `flake8 *.py`

**Comment by Aneesh02** [comment]
Still working on it!

**Comment by Aneesh02** [comment]
Hi @ZiadAmerr, I was a little occupied last week so did not work on it properly but I am having issues making the test case list as an increasing sequence. Could you please help me out with that a little?

**Comment by ZiadAmerr** [comment]
> Hi @ZiadAmerr, I was a little occupied last week so did not work on it properly but I am having issues making the test case list as an increasing sequence. Could you please help me out with that a little?

You can create a helper function that generates a list of numbers from `min_value` to `max_value`, thus generating an uniformly increasing sequence. To remove that uniformity, you can remove random elements from that sequence, this can be done by generating random integers using `st.integers` and removing the elements with indices corresponding to these generated integers. This is the simplest approach I could think of. Feel free to modify it or create your own helper function that generates an non-uniform increasing sequence.

After finishing that, use `st.one_of` to choose between an increasing sequence or a random sequence of numbers to test both cases.

**Review by ZiadAmerr — CHANGES_REQUESTED** [review]
Feel free to work on it and request a review when you're done :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by Aneesh02** [comment]
My apologies for the delay. Finally I am back working on it.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
